app/build.gradle

```diff
@@ -2,6 +2,15 @@ plugins {
   id 'com.android.application'
   id 'org.jetbrains.kotlin.android'
 }
+// Creates a variable called keystorePropertiesFile, and initializes it to the
```

Comment:
question(blocking): Since this file isn't guaranteed to exist but "should" exist for a functioning app, how should we handle this case? A hard failure on the build task is informative for local dev, but (at least currently) we also build the example app on our CI builds, which currently fails as a result of this change. Would love input/suggestions/best practices here.

Reply:
For a real production app, you're definitely on the right track. Templates for properties files are the way to go, and it's reasonable to expect programmers to copy over the files. For our case, this confers no benefit; customer keys don't make their way upstream. My best advice is to just commit a real `.properties` with "YOUR KEY HERE" or (better) "SET THIS IN local-env.properties"  placeholders.

My second-best advice is to print a warning to the logs if the file is missing, and provide default values when you're declaring your `buildConfigField`s so nothing melts. `""` would be just fine for this. Failing the build here might be seen as an unnecessary "gotcha," so please prefer to log at the [warning level](https://docs.gradle.org/current/userguide/logging.html), which I think should even surface in the IDE.